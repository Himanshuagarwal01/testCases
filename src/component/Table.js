import React, { Component } from 'react'

export default class Table extends Component {
    render() {
        return (
            <table className='custom-border'>
                <thead>
                    <tr>
                        <th>Email</th>
                        <th>Names</th>
                    </tr>
                </thead>
                <tbody>
                    <tr>
                        <td>himanshu@gmail.com</td>
                        <td>Himanshu</td>
                    </tr>
                </tbody>
            </table>
        )
    }
}
