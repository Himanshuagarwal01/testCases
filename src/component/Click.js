import React, { Component } from 'react'
import Table from './Table'

export default class Click extends Component {
    state = {
        count: 0
    }
    render() {
        return (
            <div>
                <div>
                    Count
                </div>
                <div className={`clicks-${this.state.count}`}>
                    {this.state.count} click's
                </div>
                <div>
                    <button onClick={() => this.setState({ count: this.state.count + 1 })}>Increment</button>
                </div>
                <Table />
            </div>
        )
    }
}
