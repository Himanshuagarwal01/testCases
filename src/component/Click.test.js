import Click from "./Click";
import { mount, shallow } from 'enzyme';
import Table from "./Table";
import { createRoot } from 'react-dom/client';

describe.skip('click page testing', () => {
    it('Clicks Check', () => {
        let wrapper = shallow(<Click />);
        expect(wrapper.find('.clicks-0').length).toEqual(1)
    });

    // simulate is used to click the button the page  
    it('Simulate Check', () => {
        let wrapper = shallow(<Click />);
        wrapper.find('button').simulate('click');
        expect(wrapper.find('.clicks-1').length).toEqual(1)
    });

    // dive is used to dive in to the component in the same page  
    it('Dive Check', () => {
        let wrapper = shallow(<Click />);
        wrapper.find('button').simulate('click');
        expect(wrapper.find(Table).dive().find('.custom-border').length).toEqual(1)
    });

    it('Snapshot Check', () => {
        let wrapper = shallow(<Table />);
        expect(wrapper).toMatchSnapshot();
    });
});

describe('Click Page testing', () => {
    it('Shallow vs Mount', () => {
        // let wrapper = shallow(<Click />);
        const root = createRoot(document.createElement('div'));
        root.render(<Click />);
        let wrapper = root.render(<Click />)
        console.log(wrapper);
    })
})