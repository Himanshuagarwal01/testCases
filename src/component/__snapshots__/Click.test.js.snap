// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`click page testing Snapshot Check 1`] = `
<table
  className="custom-border"
>
  <thead>
    <tr>
      <th>
        Email
      </th>
      <th>
        Names
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td>
        himanshu@gmail.com
      </td>
      <td>
        Himanshu
      </td>
    </tr>
  </tbody>
</table>
`;
